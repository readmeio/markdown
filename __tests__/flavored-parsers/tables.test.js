import { mdast } from '../../index';

describe('Parse magic block tables', () => {
  it('renders an table with missing cells', () => {
    const text = `
[block:parameters]
${JSON.stringify(
  {
    data: {
      'h-1': 'Header 1',
      '0-0': 'Cell A 1',
    },
    cols: 2,
    rows: 2,
  },
  null,
  2
)}
[/block]
    `;

    expect(mdast(text)).toMatchSnapshot();
  });
});
