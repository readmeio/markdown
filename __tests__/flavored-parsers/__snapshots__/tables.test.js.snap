// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse magic block tables renders an table with missing cells 1`] = `
Object {
  "children": Array [
    Object {
      "align": Array [
        "left",
        "left",
      ],
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "",
                },
              ],
              "type": "tableHead",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Header 1",
                },
              ],
              "type": "tableHead",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Cell A 1",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "root",
}
`;
